util/timer.h

```diff
@@ -138,8 +138,15 @@ class Timer {
   }
 
 #ifndef NDEBUG
+  // Wait until Timer starting waiting, call the optional callback, then wait
+  // for Timer waiting again.
+  // Tests can provide a custom env object to mock time, and use the callback
+  // here to bump current time and trigger Timer. See timer_test for example.
+  //
+  // Note: only support one caller of this method.
   void TEST_WaitForRun(std::function<void()> callback = nullptr) {
```

Comment:
Since Run() now releases the lock while executing the registered timer function, don't we have a race condition where this function thinks the timer is waiting but in fact it's executing the registered timer function with the mutex unlocked?

Reply:
Good point. I added `executing_task_` check here.